### PR TITLE
Meta: properly fix IPR script (#2872)

### PR DESCRIPTION
**Pull request by ljharb**
This uses the commits list API instead of the compare API, and actually covers every author. It also adds exceptions, and some legacy TODOs we need to address over time

### CODE REVIEW

File: scripts/check-form.js [hunk 1]
```diff
@@ -70,11 +74,18 @@ const request = async (url, method = 'GET', postData) => {
...
 
-const branchURL = `https://api.github.com/repos/${slug}/compare/${all ? initial : 'HEAD'}...${branch}?anon=1`;
+const perPage = 100;
+
+async function getAllCommits(page = 1) {
+	const commitsURL = `https://api.github.com/repos/${slug}/commits?anon=1&per_page=${perPage}&page=${page}&sha=${sha}`;
+	const commits = await request(commitsURL).then((json) => JSON.parse(json));
+	return [...new Set([].concat(
+		commits.flatMap(x => x?.author?.login || []),
+		(true && commits.length < perPage) ? [] : await getAllCommits(page + 1),
```
Comment (bakkot) [inline scripts/check-form.js:84]: `true &&` seems stray?

Comment (ljharb) [reply scripts/check-form.js:84]: oh whoops, thanks, i was testing

File: scripts/check-form.js [hunk 2]
```diff
@@ -128,10 +146,79 @@ const usernames = request(sheetData).then((json) => JSON.parse(json)).then(data
...
+				legacyCount -= 1;
+				console.warn(`TODO: remove ${a} from legacy list`);
+			}
+			return false;
+		}
+		return !signed;
+	});
 	if (missing.length > 0) {
-		throw `Missing authors: ${missing}`;
+		throw `Missing ${missing.length} authors: ${missing}${legacyCount > 0 ? ` \n(legacy missing: ${legacyCount})` : ''}`;
```
Comment (bakkot) [inline scripts/check-form.js:221]: I'm confused by the "legacy missing" part here. Members of the `legacy` set don't contribute to `missing.length` whether or not they've signed, so I don't know why they belong in the error message about `missing.length > 0`.

**Review by bakkot — APPROVED** [review]
Didn't run it but LGTM at a glance other than the error message I've commented on.

File: scripts/check-form.js (hunk 2 above)
Comment (ljharb) [reply scripts/check-form.js:221]: this is a way for me to see, in the case where there is someone missing, how many people *would* have been missing if not for the legacy list.

i could certainly console log it outside the "if", though ¯\\\_(ツ)_/¯

Comment (ljharb) [reply scripts/check-form.js]: k, i've fixed that

**Pull request merged**